.github/workflows/gcp-weekly.yml

```diff
@@ -0,0 +1,530 @@
+name: GCP Weekly Tests
+
+env:
+  # TODO: we really need to define a list of supported versions (ideally it's no more than 2)
+  #       build is done on the lowest version and test on the highest with a "sanity test"
+  #       stage done on all versions in the list except the highest
+  EL8_BUILD_VERSION: 8.6
+  EL8_VERSION: 8.8
+  EL9_BUILD_VERSION: 9
+  EL9_VERSION: 9
+  LEAP15_VERSION: 15.5
+  GCP_BRANCH: google/2.6
+  COVFN_DISABLED: true
+
+on:
+  schedule:
+    - cron: "0 12 * * 6"  # runs every Saturday at 12pm (UTC)
+
+# shouldn't need this, should only ever have one running at a time. But just in case
+concurrency:
+  group: gcp-weekly-${{ github.run_id }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash --noprofile --norc -ueo pipefail {0}
+
+permissions: {}
+
+jobs:
+  Create-symlinks:
+    # you might think this is an odd place to do this and it should be done as a result of the
+    # build and/or testing stages and ideally you'd be right.
+    # the problem with that is that there is no way to get the success/fail result of individual
+    # axes of matrix jobs so there is no way to query them at the end and see their composite
+    # results.
+    # instead, the final result of the Build-RPM job, for example is a last-one-complete wins.
+    # so for example, if the el9 axis fails quickly and then the el8 axis succeeds afterward the
+    # resulting job state is success.
+    # instead we assume success at the beginning and then let any axis that fails remove the
+    # lastSuccessfulBuild link if it fails
+    name: Create lastBuild and lastSuccessfulBuild symlinks
+    runs-on: [self-hosted, light]
+    env:
+      # TODO -- this should be on stable, backedup storage, not /scratch
+      # yamllint disable-line rule:line-length
+      REPO_PATH: /scratch/job_repos/daos-stack/job/daos/job/gcp-weekly-${{ github.run_id }}/
+    steps:
+      - name: Create lastBuild and lastSuccessfulBuild symlinks
+        run: . ci/gha_functions.sh;
+               mkdir -p ${REPO_PATH};
+               rm -f ${REPO_PATH}last{,Successful}Build;
+               ln -s ${{ github.run_number }} ${REPO_PATH}lastBuild;
+               ln -s ${{ github.run_number }} ${REPO_PATH}lastSuccessfulBuild
+
+  Build-RPM:
+    name: Build RPM
+    permissions:
+      statuses: write
+    runs-on: [self-hosted, docker]
+    needs: [Create-symlinks]
+    if: needs.Create-symlinks.result == 'success' &&
+        ((!cancelled()) || success() || failure())
+    strategy:
+      matrix:
+        distro: ['el8', 'el9']
+      fail-fast: false
+    env:
+      ARTIFACTORY_URL: https://artifactory.dc.hpdd.intel.com/
+      DAOS_EMAIL: brian.murrell@intel.com
+      DAOS_FULLNAME: daos-stack
+      DISTRO: ${{ matrix.distro }}
+      DISTRO_REPOS: disabled
+      DOCKER_BUILDKIT: 0
+      JENKINS_URL: https://build.hpdd.intel.com/
```

Comment:
This is going to break soon as we're moving out of the intel space. FYI @JohnMalmberg

Reply:
Is there something else I can use instead?

Reply:
Do we want to just make these GHA variables so it's only hardcoded in one place? Or as a commit pragma?

Reply:
They need to be GHA variables.

This was experimental code.

The self hosted runners are being shutdown at the old lab in one month if not sooner.
I do not know if they will be replaced at the new lab.

Yes they need to be GHA variables and there needs to be a way of getting variables from self hosted runners.

Reply:
The self hosted runners will need to be replaced somewhere or we won't be able to land stuff and/or we will be throttled by GitHub.

Reply:
Can @jolivier23 create a google/2.6 weekly testing branch (e.g., weekly-g26-testing) so we can run this in Jenkins instead?